backend/inspirehep/pidstore/providers/texkey.py

```diff
@@ -0,0 +1,149 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+import random
+import string
+
+from flask import current_app
+from inspire_utils.record import get_value
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus
+from invenio_pidstore.providers.base import BaseProvider
+from unidecode import unidecode
+
+from inspirehep.pidstore.errors import (
+    CannotGenerateUniqueTexKey,
+    NoAvailableTexKeyFound,
+    PIDAlreadyExistsError,
+    TexkeyCannotGenerateFirstPart,
+    TexkeyCannotGenerateSecondPart,
+)
+from inspirehep.records.utils import get_last_name, get_literature_earliest_date
+from inspirehep.utils import flatten_list
+
+
+class InspireTexKeyProvider(BaseProvider):
+    pid_type = "texkey"
+    pid_provider = "texkey"
+    default_status = PIDStatus.RESERVED
+
+    @classmethod
+    def sanitize(cls, name):
+        name = name.split()[0].lower()
+        name = unidecode(name)
+        name = "".join(filter(lambda x: x in set(string.ascii_lowercase), name))
```

Comment:
you should allow `-` here I think, and use all words rather than the first one only.

Reply:
:+ I will also allow digits here as I noticed that legacy allows them as well.